packages/core/src/lib/waku_message/version_0.ts

```diff
@@ -83,12 +87,13 @@ export class EncoderV0 implements Encoder {
       contentTopic: this.contentTopic,
       timestamp: BigInt(timestamp.valueOf()) * OneMillion,
       rateLimitProof: message.rateLimitProof,
+      ephemeral: this.ephemeral,
     };
   }
 }
 
 export class DecoderV0 implements Decoder<MessageV0> {
-  constructor(public contentTopic: string) {}
+  constructor(public contentTopic: string, public ephemeral: boolean = false) {}
```

Comment:
Not needed on the decoder... the decoder decodes messages, it cannot set fields on messages.

Look at how the content topic on the decoder is used in the codebase to better understand its function.